plugin/plugin.go

```diff
@@ -8,20 +8,41 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"os"
 	"strings"
 
 	"github.com/drone/drone-go/plugin/validator"
 	"github.com/open-policy-agent/opa/rego"
+	log "github.com/sirupsen/logrus"
 	yaml "gopkg.in/yaml.v2"
 )
 
 func New(policy string) validator.Plugin {
+	log.SetFormatter(&log.JSONFormatter{})
+	log.SetOutput(os.Stdout)
+
 	return &plugin{
 		policyPath: policy,
 	}
 }
 
 func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
+	promotion := Promotion{
+		Build:  req.Build.Parent,
+		Deploy: req.Build.Deploy,
+	}
+
+	// log each request for audit purposes
+	log.WithFields(log.Fields{
+		"type":       "audit",
+		"extension":  "validation",
+		"user":       req.Build.Sender,
+		"event":      req.Build.Event,
+		"repository": req.Repo.Slug,
+		"branch":     req.Repo.Branch,
+		"commit":     req.Build.After,
+		"promotion":  &promotion,
```

Comment:
**nit:** nested json a bit more annoying to work with in things like splunk.

Reply:
Let's try to filter them as separate elements and if we see that it's too complex we can flatten it.